### PR TITLE
[Doc] Improve RayService doc

### PR DESCRIPTION
**Pull request by kevin85421**
## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: docs/guidance/rayservice.md [hunk 1]
```diff
@@ -1,103 +1,115 @@
...
 > Note: This is the alpha version of Ray Services. There will be ongoing improvements for Ray Services in the future releases.
 
-### Prerequisites
+# Prerequisites
 
-* Ray 2.0 or newer.
+This guide focuses solely on the Ray Serve multi-application API, which is available starting from Ray version 2.4.0.
+
+* Ray 2.4.0 or newer.
 * KubeRay 0.6.0 or newer.
```
Comment (architkulkarni) [inline docs/guidance/rayservice.md:10]: On this line, let's specify that the nightly will work too.

File: docs/guidance/rayservice.md [hunk 2]
```diff
@@ -1,103 +1,115 @@
...
-Check that the controller is running.
+```sh
+kind create cluster --image=kindest/node:v1.23.0
+```
 
-```shell
-$ kubectl get deployments -n ray-system
-NAME           READY   UP-TO-DATE   AVAILABLE   AGE
-ray-operator   1/1     1            1           40s
+## Step 2: Install KubeRay operator
```
Comment (architkulkarni) [inline docs/guidance/rayservice.md:34]:
```suggestion
## Step 2: Install the KubeRay operator
```

File: docs/guidance/rayservice.md [hunk 3]
```diff
@@ -1,103 +1,115 @@
...
+# path: ray-operator/config/samples/
+kubectl apply -f ray_v1alpha1_rayservice.yaml
 ```
 
-List running RayServices to check on your new service `rayservice-sample`.
-```shell
-$ kubectl get rayservice
-NAME                AGE
-rayservice-sample   7s
+* Let's first take a look at the Ray Serve config (i.e. `serveConfigV2`) embedded in the RayService yaml. At a high level, there are two applications: a fruit stand app and a calculator app. Some details about the fruit stand application:
```
Comment (architkulkarni) [inline docs/guidance/rayservice.md:47]:
```suggestion
* Let's first take a look at the Ray Serve config (i.e. `serveConfigV2`) embedded in the RayService YAML. At a high level, there are two applications: a fruit stand app and a calculator app. Some details about the fruit stand application:
```

File: docs/guidance/rayservice.md [hunk 4]
```diff
@@ -1,103 +1,115 @@
...
 * **Zero downtime upgrade for Ray clusters:** Users can update the Ray cluster config in the RayService CR config and use `kubectl apply` to update the cluster. RayService will temporarily create a pending cluster and wait for it to be ready, then switch traffic to the new cluster and terminate the old one.
 * **Services HA:** RayService will monitor the Ray cluster and Serve deployments' health statuses. If RayService detects an unhealthy status for a period of time, RayService will try to create a new Ray cluster and switch traffic to the new cluster when it is ready.
 
-### Deploy the Operator
+# Example: Serve two simple Ray Serve applications using RayService
 
-Follow [this document](../../helm-chart/kuberay-operator/README.md) to install the nightly KubeRay operator via 
-Helm. Note that sample RayService in this guide uses `serveConfigV2` to specify a multi-application Serve config.
-This will be first supported in Kuberay 0.6.0, and is currently supported only on the nightly KubeRay operator.
+## Step 1: Create a Kubernetes cluster with Kind.
```
Comment (architkulkarni) [inline docs/guidance/rayservice.md:28]:
```suggestion
## Step 1: Create a Kubernetes cluster with Kind
```

**Review by architkulkarni — COMMENTED** [review]
Looks good so far, I'll take another pass once it's closer to being complete

File: docs/guidance/rayservice.md (hunk 1 above)
Comment (kevin85421) [reply docs/guidance/rayservice.md]: Updated https://github.com/ray-project/kuberay/pull/1235/commits/60ac0ef09b05b3d768939492a6d6907d6a3cbb1f

**Review by architkulkarni — APPROVED** [review]
Nice!

File: docs/guidance/rayservice.md [hunk 5]
```diff
@@ -109,149 +121,158 @@ rayservice-sample-serve-svc                        ClusterIP   10.100.39.92
...
-### Delete the RayService Instance
-```
-$ kubectl delete -f config/samples/ray_v1alpha1_rayservice.yaml
-```
+To elaborate, during the zero downtime upgrade process, RayService will create a new RayCluster temporarily and wait for it to become ready. 
+Once the new RayCluster is ready, RayService will update the selector of the head service managed by RayService (i.e., `rayservice-sample-head-svc`) to point to the new RayCluster to switch the traffic to the new RayCluster. 
+Finally, the old RayCluster will be terminated.
+
+There are certain exceptions that will not trigger a zero downtime upgrade.
+Currently, only the fields managed by Ray Autoscaler, such as `replicas` and `scaleStrategy.workersToDelete`, will not trigger a zero downtime upgrade.
```
Comment (architkulkarni) [inline docs/guidance/rayservice.md:228]: Ideally we can write down the entire list of such fields.  Are there too many to list, or are we worried that they'll frequently change?

Comment (kevin85421) [reply docs/guidance/rayservice.md:228]: `replicas` and `scaleStrategy.workersToDelete` are the only two exceptions at this moment.

Comment (kevin85421) [reply docs/guidance/rayservice.md:228]: I will merge this PR and cut a release branch. If you have any further comments, we can address them later.

**Pull request merged**